### ISSUE TITLE
Error when running gcloud container clusters get-credentials

### ISSUE THREAD

**Issue by jonathangold** [kubeflow/kubeflow#4790]
/kind bug

**What steps did you take and what happened:**

I followed the deployment guide (several times to make sure it wasn't my mistake) and the deployment fails here:
```
INFO[0297] Creating kubeflow-demo is finished: DONE      filename="gcp/gcp.go:405"
INFO[0300] Creating namespace: kubeflow                  filename="gcp/gcp.go:478"
INFO[0300] Get namespace error: namespaces "kubeflow" not found  filename="gcp/gcp.go:484"
INFO[0300] Creating namespace: istio-system              filename="gcp/gcp.go:478"
INFO[0300] Get namespace error: namespaces "istio-system" not found  filename="gcp/gcp.go:484"
INFO[0300] Binding admin role for jonathan.gold@container-solutions.com ...  filename="gcp/gcp.go:503"
INFO[0300] default-admin not found, creating...          filename="gcp/gcp.go:537"
INFO[0300] Running get-credentials to build .kubeconfig  filename="gcp/gcp.go:797"
INFO[0300] Running get-credentials kubeflow-demo --zone=us-east1-c --project=jon-ml-ops ...  filename="gcp/gcp.go:804"
Fetching cluster endpoint and auth data.
ERROR: (gcloud.container.clusters.get-credentials) get-credentials requires edit permission on jon-ml-ops
Error: failed to apply:  (kubeflow.error): Code 500 with message: coordinator Apply failed for gcp:  (kubeflow.error): Code 400 with message: gcp apply could not update deployment manager Error Error when running gcloud container clusters get-credentials: exit status 1
```

**What did you expect to happen:**
I expected the deployment to work :)

**Anything else you would like to add:**

I also tried running `gcloud container clusters get-credentials kubeflow-demo` but get the same error:
`ERROR: (gcloud.container.clusters.get-credentials) get-credentials requires edit permission on jon-ml-ops`
Same behaviour with other service accounts.

**Environment:**

- Kubeflow version: (version number can be found at the bottom left corner of the Kubeflow dashboard): N/A
- kfctl version: (use `kfctl version`): kfctl v1.0-rc.3-1-g24b60e8
- Kubernetes platform: (e.g. `minikube`) GKE
- Kubernetes version: (use `kubectl version`): Client v1.15.2 (no server at the moment)
- OS (e.g. from `/etc/os-release`): Fedora 31 Workstation Edition

**Comment by jlewi** [comment]
Do you get the same behavior if your run the gcloud command directly? 

You can run `gcloud auth list` to see which account cloud is authorized as. gcloud is most likely using an account that doesn't have the appropriate permission and that is why you are getting the permission error.

If you want to use a service account with gcloud make sure you have activated the service account.
https://cloud.google.com/sdk/gcloud/reference/auth/activate-service-account

**Comment by jonathangold** [comment]
Yes, I ran the command directly using both my "Owner" account as well as a service account with all the required permissions. The only 'unconventional' thing I can think of is that I'm using a gsuite email for my google account.

**Comment by Shaked** [comment]
@jonathangold have you figured it out? 

This seems to be related to the service account's roles. 

For some reason, the deployment (or upgrade) expect the service account to have permissions instead of using the user that logged in via `gcloud auth login`

**Comment by emilecaron** [comment]
I'm having similar issue here. What specific permission should be set on the service account ?

**Comment by jlewi** [comment]
You should run 

```
gcloud auth list
```

To see which account is activated and being used. To deploy Kubeflow you need project OWNER permissions. 

To get-credentials just needs container-clusterViewer role
https://cloud.google.com/kubernetes-engine/docs/how-to/iam#predefined

But that will be insufficient for deploying Kubeflow.

**Comment by jlewi** [comment]
kfctl uses a combination of GCP go client libraries and shelling out to gcloud. I think the latter is just used to setup a KUBECONFIG entry.

The client libraries depend on the application default credentials. If you want to use your end user account credentials for application default you need to login using

``
gcloud auth application-default login
```
